app/controllers/dashboard_controller.rb

```diff
@@ -2,8 +2,8 @@ class DashboardController < ApplicationController
   include Foreman::Controller::AutoCompleteSearch
   include Foreman::Controller::Parameters::Widget
 
-  before_action :prefetch_data, :only => :index
-  before_action :find_resource, :only => [:destroy]
+  before_action :prefetch_data, :only => :show
```

Comment:
This isn't ideal, as the same data will be loaded for each widget. In the next step this should be refactored so each widget only loads the data it actually needs, possibly using cache to prevent the same data from being loaded twice.